distsql/request_builder.go

```diff
@@ -125,7 +125,13 @@ func (builder *RequestBuilder) SetKeepOrder(order bool) *RequestBuilder {
 }
 
 func (builder *RequestBuilder) getIsolationLevel(sv *variable.SessionVars) kv.IsoLevel {
-	isoLevel, _ := sv.GetSystemVar(variable.TxnIsolation)
+	var isoLevel string
+	if sv.TxnIsolationLevelOneShot.State == 2 {
```

Comment:
Why does it need special handling？

Reply:
> set statement still run into doCommitWithRetry, if I don't maintain a state for tx_isolation_one_shot, it will be reset immediately, in the set statement itself. 
@zimulala